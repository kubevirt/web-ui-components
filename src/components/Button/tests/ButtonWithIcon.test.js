import createTestContext from '../../../cosmos/enzyme';
import fixture from '../fixtures/ButtonWithIcon.fixture';

const { mount, getWrapper } = createTestContext({ fixture });

beforeEach(mount);

test('renders Button With Icon', () => {
  expect(getWrapper().text()).toEqual('Button With Icon');
});

test('matches snapshot', () => {
  expect(getWrapper()).toMatchSnapshot();
});
