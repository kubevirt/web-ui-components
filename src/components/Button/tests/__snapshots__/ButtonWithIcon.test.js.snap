// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ButtonWithIcon
  icon="virtual-machine"
  iconType="pf"
  label="Button With Icon"
  onClick={[MockFunction]}
>
  <Button
    active={false}
    block={false}
    bsClass="btn btn-link btn-with-icon"
    bsStyle="default"
    disabled={false}
    onClick={[MockFunction]}
  >
    <button
      className="btn btn-link btn-with-icon btn btn-link btn-with-icon-default"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={12}
      >
        <div
          className="col-md-12"
        >
          <Icon
            className="fa-5x"
            name="virtual-machine"
            type="pf"
          >
            <PatternflyIcon
              className="fa-5x"
              name="virtual-machine"
            >
              <span
                aria-hidden="true"
                className="pficon pficon-virtual-machine fa-5x"
              />
            </PatternflyIcon>
          </Icon>
        </div>
      </Col>
      <Col
        bsClass="col"
        className="label-column"
        componentClass="div"
        md={12}
      >
        <div
          className="label-column col-md-12"
        >
          <span
            className="lead"
          >
            Button With Icon
          </span>
        </div>
      </Col>
    </button>
  </Button>
</ButtonWithIcon>
`;
